Formula/sonic.rb

```diff
@@ -0,0 +1,60 @@
+class Sonic < Formula
+  desc "Fast, lightweight & schema-less search backend"
+  homepage "https://github.com/valeriansaliou/sonic"
+  url "https://github.com/valeriansaliou/sonic/archive/v1.2.4.tar.gz"
+  sha256 "290da969d39260abe5772fc7f084d256a8c0a53c702fe9eb44593d43a179b9d1"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
+    inreplace "config.cfg", "./", var/"sonic/"
+    etc.install "config.cfg" => "sonic.cfg"
+  end
+
+  plist_options :manual => "sonic -c #{HOMEBREW_PREFIX}/etc/sonic.cfg"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>KeepAlive</key>
+          <dict>
+            <key>SuccessfulExit</key>
+            <false/>
+          </dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{opt_bin}/sonic</string>
+            <string>-c</string>
+            <string>#{etc}/sonic.cfg</string>
+          </array>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>WorkingDirectory</key>
+          <string>#{var}</string>
+          <key>StandardErrorPath</key>
+          <string>#{var}/log/sonic.log</string>
+          <key>StandardOutPath</key>
+          <string>#{var}/log/sonic.log</string>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    port = free_port
+
+    cp etc/"sonic.cfg", testpath/"config.cfg"
+    inreplace "config.cfg", ":1491", ":#{port}"
+    inreplace "config.cfg", "#{var}/sonic", "."
+
+    fork { exec bin/"sonic" }
+    sleep 2
+    system "nc", "-z", "localhost", port
```

Comment:
Is there no query you can run against the server?

Reply:
Sorry, meant to mention that in notes. It uses its own wire protocol.

Reply:
Ref: https://github.com/valeriansaliou/sonic#perform-searches-and-manage-objects